controllers/openstackdataplanenode_controller.go

```diff
@@ -199,12 +199,27 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 
 	r.Log.Info("Node", "DeployStrategy", instance.Spec.DeployStrategy.Deploy, "Node.Namespace", instance.Namespace, "Node.Name", instance.Name)
 	if instance.Spec.DeployStrategy.Deploy {
+
+		// create deployIdentifier
+		if len(instance.Spec.DeployStrategy.DeployIdentifier) == 0 {
+			// patch := client.MergeFrom(instance.DeepCopy())
+			newIdentifier := dataplanev1beta1.GenerateDeployIdentifier()
+			instance.Spec.DeployStrategy.DeployIdentifier = newIdentifier
```

Comment:
This is why it keeps reconciling, because we update the spec here. Each time we reach here, we call `GenerateDeployIdentifier()` which gives us a new unique string, and causes us to reconcile again. Let's update the status instead with the current identifier. I think we need some additional logic here to protect against generating a new string each reconcile loop, since the operator will reconcile by default every 10 minutes, the current implementation would result in a new identifier every 10 minutes, potentially triggering an action.

It probably makes sense to have a number that increments rather than a randomly generated string? So if we update this to first check the current deploy identifier, if it's "", then we need to generate one. If it ISN'T "", then we should increment it if required. So, if the instance Spec has changed, then we need to increment it for example.

Reply:
In this if statement here, we're checking to see IF the user has provided us a custom DeployIdentifier, if they have NOT, then we are generating a new one. Expanding on my above point, I think we should only do this if one doesn't exist in the Status as well. So the options here are,

1. Instead of a string, we could return a int64 from the `GenerateDeployIdentifier()` method. Then we could increment that integer for each subsequent deploy. (But maybe we don't care to be that granular, and all we really want is uniqueness? If so, see option 2)

2. We simply change this to: `instance.Status.DeployIdentifier = newIdentifier`. Although, it would probably be more efficient to just do:

```
instance.Status.DeployIdentifier = dataplanev1beta1.GenerateDeployIdentifier()
```

Then we save space on the variable allocation. 

In cases where a use HAS provided a DeployIdentifier, then we want to introduce a new method with logic that will determine what action we should take in that case. So, I think what we would want to do is to set the `instance.Status.DeployIdentifier == instance.Spec.DeployStrategy.DeployIdentifier`. 

Since the deploy Identifier has changed, we then need to take action based upon that outcome and recreate the jobs.

Reply:
> This is why it keeps reconciling, because we update the spec here. Each time we reach here, we call `GenerateDeployIdentifier()` which gives us a new unique string, and causes us to reconcile again. Let's update the status instead with the current identifier. I think we need some additional logic here to protect against generating a new string each reconcile loop, since the operator will reconcile by default every 10 minutes, the current implementation would result in a new identifier every 10 minutes, potentially triggering an action.

The GenerateDeployIdentifier is only reached if the current length on the spec is 0, so I don't think it would continually reconcile past the first time.
 
> It probably makes sense to have a number that increments rather than a randomly generated string? So if we update this to first check the current deploy identifier, if it's "", then we need to generate one. If it ISN'T "", then we should increment it if required. So, if the instance Spec has changed, then we need to increment it for example.

What would be the difference between a random/unique value and an incrementing value?

I agree that setting instance.Status.DeployIdentifier would be useful, and we could use that to track if the deployment "in progress" is the one started for the requested deploy.